packages/shared/forks/ReactFeatureFlags.native-oss.js

```diff
@@ -10,100 +10,110 @@
 import typeof * as FeatureFlagsType from 'shared/ReactFeatureFlags';
 import typeof * as ExportsType from './ReactFeatureFlags.native-oss';
 
+// TODO: Align these flags with canary and delete this file once RN ships from Canary.
+
 // -----------------------------------------------------------------------------
-// Ready for next major.
+// TODO for next React Native major.
 //
-// Alias __NEXT_MAJOR__ to false for easier skimming.
+// Alias __TODO_NEXT_RN_MAJOR__ to false for easier skimming.
 // -----------------------------------------------------------------------------
-const __NEXT_MAJOR__ = false;
```

Comment:
Just to clarify: the original meaning of this variable when I added was "next React major", not "next RN major". The reason for it was that I needed to gate a feature because npm's React wouldn't define the cache object in the secret internals object, and enabling the cache feature in the RN renderer would throw an error, even if unused. When we release a new version of React we can pin RN to it and this problem would go away, which means we would be able to enable cache support.

Reply:
Yeah but these flags do not turn on/off anything in the React build, only the renderers. So calling it the "next react" major is wrong, it's really "the next react native renderer major" which can be after the next react npm, major. This leaves the option open to shipping RN renderer fixes in the time period between when React ships to npm and React Native upgrades to the lastest React version.

Once RN uses the canary then "next react major" === "next react native renderer major"

Reply:
The intent isn't to enable anything on React but to define what's the dependency. This cannot be turned on until there's a release of React (not of React Native). We could have new "majors" of RN without synchronizing the renderer and we wouldn't change this value yet. It's only when there's a new release of React to npm and we update the dependency in RN that we can change it.

Reply:
Right, but when we cut a React release, I would expect to be able to grep all `__NEXT_MAJOR__` instances and replace them with true. That is not the case for these flags, so they need a different name. Maybe "NEXT RN MAJOR" isn't correct, but it's better than before.

Reply:
Why is it not the case with these flags? I think in that scenario it would be safe to do it. Then when you sync the renderer to the RN repository you just need to bump the React version and commit it together.

Reply:
If you wanted to ship a bugfix (or type fixes) in the renderers without upgrading to the latest React, you couldn't.

Reply:
That's right, but in the places where we used `__TODO_NEXT_RN_MAJOR__` you wouldn't be able to do it either (that's exactly what happened with the flag for using React cache).